test/extended/openstack/egressip.go

```diff
@@ -0,0 +1,470 @@
+		e2e.Logf("Found '%s' free IP address in the EgressIP network in Openstack", egressIPAddrStr)
+
+		g.By("Creating a temp directory")
+		egressIPTempDir, err := os.MkdirTemp("", "egressip-e2e")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		e2e.Logf("Created '%s' temporary directory", egressIPTempDir)
+		defer os.RemoveAll(egressIPTempDir)
+
+		g.By("Creating an EgressIP yaml file")
+		var egressIPname = "egress-ip"
+		var egressIPYamlFileName = "egressip.yaml"
+		var egressIPYamlFilePath = egressIPTempDir + "/" + egressIPYamlFileName
+		var egressIPYamlTemplate = `apiVersion: k8s.ovn.org/v1
+kind: EgressIP
+metadata:
+  name: %s
+spec:
+  egressIPs:
+  - %s
+  namespaceSelector:
+    matchLabels:
+      %s`
+
+		egressIPYaml := fmt.Sprintf(egressIPYamlTemplate, egressIPname, egressIPAddrStr, "app: egress")
+		e2e.Logf("egressIPYaml: %s", egressIPYaml)
+
+		err = os.WriteFile(egressIPYamlFilePath, []byte(egressIPYaml), 0644)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By(fmt.Sprintf("Creating an EgressIP object from '%s'", egressIPYamlFilePath))
+		err = oc.AsAdmin().Run("create").Args("-f", egressIPYamlFilePath).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		defer oc.AsAdmin().Run("delete").Args("egressip", egressIPname).Execute()
+
+		g.By("Waiting until CloudPrivateIPConfig is created and assigned to the primary worker node")
+		cloudNetworkClientset, err = cloudnetwork.NewForConfig(oc.AdminConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		waitOk, err := waitCloudPrivateIPConfigAssignedNode(cloudNetworkClientset, egressIPAddrStr, primaryWorker.Name)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(waitOk).To(o.BeTrue(), "Not found the expected assigned node '%s' in '%s' CloudPrivateIPConfig", primaryWorker.Name, egressIPAddrStr)
+		e2e.Logf("Found the expected assigned node '%s' in '%s' CloudPrivateIPConfig", primaryWorker.Name, egressIPAddrStr)
+
+		// Label the secondary worker node as EgressIP assignable node
+		g.By(fmt.Sprintf("Labeling the secondary node '%s' with '%s'", secondaryWorker.Name, egressAssignableLabelKey))
+		e2e.AddOrUpdateLabelOnNode(clientSet, secondaryWorker.Name, egressAssignableLabelKey, "dummy")
+		defer e2e.RemoveLabelOffNode(clientSet, secondaryWorker.Name, egressAssignableLabelKey)
+
+		// Check the assigned node in CloudPrivateIPConfig is kept with the primary EgressIP worker node
+		g.By("Checking the CloudPrivateIPConfig object and the assigned node after adding a second EgressIP assignable node")
+		cpicAssignedNode, err := getCpipAssignedNode(cloudNetworkClientset, egressIPAddrStr)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Could not find the assigned node in '%s' CloudPrivateIPConfig", egressIPAddrStr)
+		e2e.Logf("'%s' CloudPrivateIPConfig assigned node ('%s') is kept after adding a second assignable node", egressIPAddrStr, cpicAssignedNode)
+		o.Expect(cpicAssignedNode).To(o.Equal(primaryWorker.Name), "'%s' egressIP has been deassigned to the '%s' node", egressIPAddrStr, primaryWorker.Name)
```

Comment:
Do we have other tests doing some Machine operations like a reboot? I wonder if that wouldn't interfere here